pkg/deploy/strategy/recreate/recreate.go

```diff
@@ -88,6 +93,25 @@ func (s *RecreateDeploymentStrategy) Deploy(from *kapi.ReplicationController, to
 	return s.DeployWithAcceptor(from, to, desiredReplicas, nil)
 }
 
+// recordControllerWarnings records the replication controller warnings into a
+// deployment event sink.
+func (s *RecreateDeploymentStrategy) recordControllerWarnings(rc *kapi.ReplicationController) {
```

Comment:
How about passing the eventsClient, errOut, and the decoder as arguments for this to be a standalone function so you can reuse it between the strategies? You probably don't need the `s.events == nil` too